rest/data/commit_queue.go

```diff
@@ -280,6 +283,103 @@ func (pc *DBCommitQueueConnector) IsAuthorizedToPatchAndMerge(ctx context.Contex
 	return hasPermission, nil
 }
 
+// EnqueuePRToCommitQueue enqueues an existing pull request to the commit queue.
+func EnqueuePRToCommitQueue(ctx context.Context, sc Connector, settings *evergreen.Settings, info commitqueue.PRInfo) (*restModel.APIPatch, error) {
+	userRepo := UserRepoInfo{
+		Username: info.Username,
+		Owner:    info.Owner,
+		Repo:     info.Repo,
+	}
+	authorized, err := sc.IsAuthorizedToPatchAndMerge(ctx, settings, userRepo)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting user info from GitHub API")
+	}
+	if !authorized {
+		return nil, errors.Errorf("user '%s' is not authorized to merge", userRepo.Username)
+	}
+
+	prNum := info.PR
```

Comment:
nit: It seems like the original logic only assigned this to a new variable because it had to dereference the GitHub event pointer. Can we just use `info.PR` in all the places that need it rather than making a new local variable?